Firebase/Core/FIRComponentContainer.m

```diff
@@ -161,14 +171,16 @@ - (nullable id)instanceForProtocol:(Protocol *)protocol {
 
 - (void)removeAllCachedInstances {
   // Loop through the cache and notify each instance that is a maintainer to clean up after itself.
-  for (id instance in self.cachedInstances.allValues) {
-    if ([instance conformsToProtocol:@protocol(FIRComponentLifecycleMaintainer)] &&
-        [instance respondsToSelector:@selector(appWillBeDeleted:)]) {
-      [instance appWillBeDeleted:self.app];
+  dispatch_sync(_containerQueue, ^{
+    for (id instance in self.cachedInstances.allValues) {
+      if ([instance conformsToProtocol:@protocol(FIRComponentLifecycleMaintainer)] &&
+          [instance respondsToSelector:@selector(appWillBeDeleted:)]) {
+        [instance appWillBeDeleted:self.app];
```

Comment:
Nit: Do you think it may somehow call protected by `_containerQueue`? If yes, then we may use separate `dispatch_sync` to copy `self.cachedInstances` and for `[self.cachedInstances removeAllObjects];`

Reply:
I don't think so but it's probably a good idea regardless. Will look at doing so.